dask/dataframe/categorical.py

```diff
@@ -58,30 +74,63 @@ def categorize(df, columns=None, index=None, **kwargs):
         Whether to categorize the index. By default, object indices are
         converted to categorical, and unknown categorical indices are made
         known. Set True to always categorize the index, False to never.
+    split_every : int, optional
+        Group partitions into groups of this size while performing a
+        tree-reduction. If set to False, no tree-reduction will be used.
+        Default is False.
```

Comment:
A small benchmark showed 16 to be a pretty good number, so I went with that. This of course is heavily dependent on number of categoricals and partitions.